### ISSUE TITLE
Angular Universal: '.setElementClass' failing when being called with 'undefined' as className

### ISSUE THREAD

**Issue by florianmrz** [sherweb/ngx-materialize#329]
The two icon directives `mz-icon` and `mz-icon-mdi` both call the `.setElementClass` function of the `Renderer` with a `className` value of `undefined`. This will not cause any issues in the browser as raising an error would be [against the spec](https://dom.spec.whatwg.org/#dom-domtokenlist-remove), but due to Angular Universal pre-rendering HTML on the server using the [domino package](https://github.com/fgnass/domino), it causes things to break.

## Current Behavior
Attempting to pre-render an Angular app on the server throws an error when trying to add / remove a class that is `undefined`, despite this being [within the spec](https://dom.spec.whatwg.org/#dom-domtokenlist-remove). This is most likely due to Angular using the [domino package](https://github.com/fgnass/domino) to create the DOM in NodeJS that doesn't allow the input of `undefined` as className while adding / removing a CSS class to an element:

Error handler that throws the error [(Source)](https://github.com/fgnass/domino/blob/a886e56d4e3bd3afa267883d1777570c108c6e4b/lib/DOMTokenList.js#L80-L82)
```javascript
function handleErrors(token) {
  if (token === "" || token === undefined) {
    utils.SyntaxError();
  }
  ...
}
```

There are a three places where this is causing issues:
- [MzIconDirective.handleAlign](https://github.com/sherweb/ng2-materialize/blob/master/src/app/icon/icon.directive.ts#L45) (`this.align` is initially `undefined`)
- [MzIconDirective.handleSize](https://github.com/sherweb/ng2-materialize/blob/master/src/app/icon/icon.directive.ts#L56) (`this.size` is initially `undefined`)
- [MzIconMdiDirective.handleAlign](https://github.com/sherweb/ng2-materialize/blob/master/src/app/icon-mdi/icon-mdi.directive.ts#L49) (`this.align` is initially `undefined`)

## Possible Solution

Current:
```typescript
handleAlign(previousValue?: string) {
  if (previousValue) {
    this.renderer.setElementClass(this.elementRef.nativeElement, previousValue, false);
  }
  this.renderer.setElementClass(this.elementRef.nativeElement, this.align, !!this.align);
}
```
Wrap the method inside of an `if` statement to make sure we are not adding / removing a class that is `undefined`:
```typescript
handleAlign(previousValue?: string) {
  if (previousValue) {
    this.renderer.setElementClass(this.elementRef.nativeElement, previousValue, false);
  }
  if (this.align) {
    this.renderer.setElementClass(this.elementRef.nativeElement, this.align, true);
  }
}
```

I know this is not really a bug in the ng2-materialize library, but seeing that this is such an easy fix and probably not intended in the first place, I would love to create a PR for this.

## Steps to Reproduce (for bugs)
Sadly I cannot provide a fiddle to reproduce this as this only affects Angular Universal, where the app is being rendered by the server and not the browser, and I wasn't able to set one up for that.
I'm using the pre-render strategy mentioned in the [Angular universal starter kit](https://github.com/angular/universal-starter), running Angular 6.0.0. 

## Your Environment

* Version used: 1.8.1
* Angular version used: 6.0.0
* Browser Name and version: Server side (NodeJS v8.9.1)

**Comment by jfcere** [comment]
This is strange because `!!undefined` will result in `false` so the value `undefined` should never been set as class name ... are you sure about this?

**Comment by florianmrz** [comment]
That's correct, but this it the third parameter that determines whether or not the class should be added or removed, the second one is the className. In this case this would result in:
```typescript
.setElementClass(native element, undefined, false)

```
The DOM Renderer will therefore try to **remove** the class `undefined` from the element, which causes the issue.

**Comment by jfcere** [comment]
Oh jeez ... I focused on the last one ... you're right!

**Comment by jfcere** [comment]
Yes you can provide the PR that would be really appreciated, but we are about to release v6.0.0 with major changes to the project architecture so this might be a little painful to merge.

**Comment by florianmrz** [comment]
OK, then I will wait until this is out and provide the PR afterwards, I implemented a workaround on my end and I don't think there's too many people affected by it anyways.

**Comment by jfcere** [comment]
Update is done, you can submit your PR now

**Comment by florianmrz** [comment]
Awesome, while rewriting this part, would you like me to also update the used renderer to `Renderer2`?
I've seen it being used in the following module, but also read something about keeping the old one due to backwards compatibility:
https://github.com/sherweb/ngx-materialize/blob/bee572cea032fdc9107aad7fe5b4be1600e67f97/lib/src/collection/collection.component.ts#L8-L15

**Comment by jfcere** [comment]
There is a lot of tests to correct if replacing for `Renderer2`. We were planing to remove it when migrating to Materialize-Css v1.0 since we will have to touch pretty much everything to remove JQuery and use plain JavaScript instead.

**Comment by jfcere** [comment]
Approved and merged... thanks again for your contribution.
I will do a v6.0.1 release for your changes tonight!

**Comment by florianmrz** [comment]
Thanks for giving me the opportunity to help out!

**Comment by jfcere** [comment]
Your fix has been released, you can update to [v6.0.1](https://github.com/sherweb/ngx-materialize/releases/tag/v6.0.1)
